File: dotnet/ArqLibrarianClassic/ArcLibrarianClassis.ATests/SpyUserIn.cs
using System.Collections.Generic;
using ArqLibrarianClassic;

namespace ArcLibrarianClassis.ATests
{
    public class SpyUserIn : UserIn
    {
        private readonly InputAware input;
        private readonly Queue<string> entered = new Queue<string>();

        public SpyUserIn(InputAware input)
        {
            this.input = input;
        }

        internal void EnterLine(string text)
        {
            entered.Enqueue(text);
            input.OnTextLineEntered(text);
        }

        public string ReadLine()
        {
            return entered.Dequeue();
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayeredATests/src/bnsit/ms/arq/SpyUserIn.java
package bnsit.ms.arq;

import java.util.LinkedList;
import java.util.Queue;

public class SpyUserIn implements UserIn {
    private final InputAware input;
    private final Queue<String> entered = new LinkedList<String>();

    public SpyUserIn(InputAware input)
    {
        this.input = input;
    }

    public void enterLine(String text) {
        entered.add(text);
        input.onTextLineEntered(text);
    }

    @Override
    public String readLine() {
        return entered.poll();
    }
}
